data/awsrds/v0/main.go

```diff
@@ -0,0 +1,448 @@
+
+	// Test every engines to find the correct one
+	var db *sqlx.DB
+	var err error
+	for _, engine := range engines {
+
+		dsn := fmt.Sprintf(engine,
+			cfg.DBUser, cfg.DBPassword, DBEndpoint, cfg.DBName,
+		)
+
+		db, err = sqlx.Open("mysql", dsn)
+		if err == nil {
+			return db, nil
+		}
+	}
+
+	return nil, fmt.Errorf("error creating SQL client")
+}
+
+// newClientIAM creates a new SQL client with IAM authentication
+func newClientIAM(e *execution) (*sqlx.DB, error) {
+	cfg := LoadConfig(e)
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(cfg.Region),
+		Credentials: credentials.NewStaticCredentials(
+			cfg.DBAWSAccessKeyID, cfg.DBAWSSecretAccessKey, ""),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	DBEndpoint := fmt.Sprintf("%s:%s", cfg.DBHost, cfg.DBPort)
+
+	authToken, err := rdsutils.BuildAuthToken(
+		DBEndpoint,
+		cfg.Region,
+		cfg.DBUser,
+		sess.Config.Credentials,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Test every engines to find the correct one
+	var db *sqlx.DB
+	for _, engine := range engines {
+		dsn := fmt.Sprintf(engine+"?tls=true&tls=skip-verify&allowCleartextPasswords=true",
+			cfg.DBUser, authToken, DBEndpoint, cfg.DBName,
+		)
+
+		db, err = sqlx.Open("mysql", dsn)
+		if err == nil {
+			return db, nil
+		}
+	}
+
+	return nil, fmt.Errorf("error creating SQL client")
+}
+
+func getUser(setup *structpb.Struct) string {
+	return setup.GetFields()["user"].GetStringValue()
+}
+func getPassword(setup *structpb.Struct) string {
+	return setup.GetFields()["password"].GetStringValue()
+}
+func getName(setup *structpb.Struct) string {
+	return setup.GetFields()["name"].GetStringValue()
+}
+func getHost(setup *structpb.Struct) string {
+	return setup.GetFields()["host"].GetStringValue()
+}
+func getPort(setup *structpb.Struct) string {
+	port := setup.GetFields()["port"].GetNumberValue()
+	portStr := strconv.FormatFloat(port, 'f', -1, 64)
+	return portStr
+}
+func getRegion(setup *structpb.Struct) string {
+	return setup.GetFields()["region"].GetStringValue()
+}
+func getAWSAccessKeyID(setup *structpb.Struct) string {
+	return setup.GetFields()["aws-access-key-id"].GetStringValue()
+}
+func getAWSSecretAccessKey(setup *structpb.Struct) string {
+	return setup.GetFields()["aws-secret-access-key"].GetStringValue()
+}
+
+func (e *execution) Execute(ctx context.Context, inputs []*structpb.Struct) ([]*structpb.Struct, error) {
+	outputs := []*structpb.Struct{}
+
+	var client *sqlx.DB
+	var err error
+
+	// check if the password field is present, if it is, create a client with username and password else create a client with IAM
+	if e.Setup.GetFields()["password"] != nil {
+		client, err = newClient(e)
+	} else if e.Setup.GetFields()["password"] == nil && e.Setup.GetFields()["aws-access-key-id"] != nil && e.Setup.GetFields()["aws-secret-access-key"] != nil {
+		client, err = newClientIAM(e)
+	} else {
+		return nil, fmt.Errorf("some fields are missing, please fill in the required fields")
+	}
```

Comment:
Question
```
	// check if the password field is present, if it is, create a client with username and password else create a client with IAM
	if e.Setup.GetFields()["password"] != nil {
		client, err = newClient(e)
	} else if e.Setup.GetFields()["password"] == nil && e.Setup.GetFields()["aws-access-key-id"] != nil && e.Setup.GetFields()["aws-secret-access-key"] != nil {
		client, err = newClientIAM(e)
	} else {
		return nil, fmt.Errorf("some fields are missing, please fill in the required fields")
	}
```

Should this logic include to newClient?
e.g.
newClient: check if the users input a password
  - if yes -> newTraditionalClient (Probably need to tune the name)
  - if no -> newIAMClient (Probably need to tune the name)

Reply:
@chuang8511 

You mean to put the conditional logic within newClient and make two functions based on the authentication method the user chooses? That would also work.

Reply:
> conditional logic within newClient and make two functions based on the authentication method the user chooses?

Yes! but actually it is only some preference to make code explainable.